### PR TITLE
Make Vector::values a std::unique_ptr.

### PR DESCRIPTION
**Pull request by bangerth**
The only tricky part are the changes in `VectorView`, with its quirky semantics (see also #5278).

### CODE REVIEW

File: include/deal.II/lac/vector.templates.h [hunk 1]
```diff
@@ -1010,7 +1011,7 @@ template <typename Number>
 void
 Vector<Number>::deallocate()
 {
-  free(values);
+  free(values.get());
```
Comment (masterleinad) [inline include/deal.II/lac/vector.templates.h]: Doesn't this (and the line below) call `free` twice? I guess `values.reset()` should be sufficient here.

Comment (bangerth) [reply include/deal.II/lac/vector.templates.h]: Uh, oh. You are right. Let me go back to this and fix up a couple of other places where I can simplify things (though they are not wrong).

**Comment by bangerth** [comment]
OK, this should be better!

**Comment by masterleinad** [comment]
All these calls to `std::unique_ptr::release()` without using the return value make me shiver. :wink:

**Comment by bangerth** [comment]
I agree. That's why I want to get rid of that class!

**Pull request merged**